google/cloud/pubsublite/internal/partition_publisher.cc

```diff
@@ -70,7 +73,8 @@ future<StatusOr<Cursor>> PartitionPublisher::Publish(PubSubMessage m) {
   if (!status.ok()) {
     return make_ready_future(StatusOr<Cursor>(std::move(status)));
   }
-  MessageWithPromise unbatched{std::move(m), promise<StatusOr<Cursor>>{}};
+  MessageWithPromise unbatched{std::move(m), next_sequence_++,
```

Comment:
Is there a reason to use post-fix ++?  If so, please add a comment, we rarely use them.